trusted/crypto/src/lib.rs

```diff
@@ -16,17 +19,33 @@ impl From<TryFromIntError> for Error {
     }
 }
 
+/// A SHA256 hash over some data.  See [Sha256::hash()] and [Sha256Builder]
 pub struct Sha256 {
     hash: [u8; Sha256::SIZE],
 }
 
 impl Sha256 {
-    pub const SIZE: usize = 32;
+    const SIZE: usize = 32;
 
+    /// The hash value as a byte slice
     pub fn as_slice(&self) -> &[u8] {
         &self.hash
     }
 
+    /// Compute the hash of `data`.
```

Comment:
With the addition of the `Builder` would it make sense to rename this `from_bytes` or similar?

cc: @remoun since you already looked over this once